task4/controllers/productsController.js

```diff
@@ -0,0 +1,13 @@
+const getProducts = (req, res, products) => {
+  res.json(products || []);
+};
+
+const getProductById = (req, res) => {
+  if (!req.product) {
+    res.send('Invalid User id!');
```

Comment:
https://expressjs.com/en/guide/error-handling.html

It would be better send 400 status code